blurring_as_a_service/inference_pipeline/components/detect_and_blur_sensitive_data.py

```diff
@@ -90,47 +95,65 @@ def detect_and_blur_sensitive_data(
         raise FileNotFoundError(f"The folder '{batches_files_path}' does not exist.")
     # Iterate over files in the folder
     for batch_file_txt in os.listdir(batches_files_path):
-        file_path = os.path.join(batches_files_path, batch_file_txt)
-        # Check if the path points to a file (not a directory)
-        if os.path.isfile(file_path):
-            print(f"Creating inference step: {file_path}")
-
-            files_to_blur_full_path = os.path.join(
-                yolo_yaml_path, batch_file_txt
-            )  # use outputs folder as Azure expects outputs there
-            with open(file_path, "r") as src:
-                with open(files_to_blur_full_path, "w") as dest:
-                    for line in src:
-                        dest.write(f"{input_structured_folder}/{line}")
-                        print(f"{input_structured_folder}/{line}")
-
-            data = dict(
-                train=f"{files_to_blur_full_path}",
-                val=f"{files_to_blur_full_path}",
-                test=f"{files_to_blur_full_path}",
-                nc=2,
-                names=["person", "license_plate"],
-            )
-
-            # TODO create postgresql string and send to val.py
-
-            with open(f"{yolo_yaml_path}/pano.yaml", "w") as outfile:
-                yaml.dump(data, outfile, default_flow_style=False)
-            cuda_device = torch.cuda.current_device()
-            model_parameters = json.loads(model_parameters_json)
-            database_parameters = json.loads(database_parameters_json)
-            val.run(
-                weights=model,
-                data=f"{yolo_yaml_path}/pano.yaml",
-                project=results_path,
-                device=cuda_device,
-                name="",
-                customer_name=customer_name,
-                start_time=get_current_time(),
-                run_id=generate_unique_string(10),
-                **model_parameters,
-                **database_parameters,
-            )
-            delete_file(file_path)
-            delete_file(files_to_blur_full_path)
-            delete_file(f"{yolo_yaml_path}/pano.yaml")
+        if batch_file_txt.endswith('.txt'):
+            file_path = os.path.join(batches_files_path, batch_file_txt)
+
+            # Check if the path points to a file (not a directory) and if the file exists
+            if os.path.isfile(file_path) and os.path.exists(file_path):
+                logger.info(f"Creating inference step: {file_path}")
+
+                files_to_blur_full_path = os.path.join(
+                    yolo_yaml_path, batch_file_txt
+                )  # use outputs folder as Azure expects outputs there
+
+                # Define the locked file path with ".lock" extension
+                locked_file_path = file_path + ".lock"
+
+                read_success = False
+                try:
+                    # Rename the file to add ".lock" extension
+                    os.rename(file_path, locked_file_path)
```

Comment:
I was thinking that using a generic LockFile object will make all the code cleaner and the function reusable in the future. 
Here is an example that has not been tested:

```
class LockFile:
    def __init__(self, file_path):
        self.file_path = file_path
        self.locked_file_path = self.file_path + ".lock"

    def __enter__(self):
        try:
            os.rename(self.file_path, self.locked_file_path)

            with open(self.locked_file_path, "r") as src:
                return src
        except FileNotFoundError as e:
            logger.info(f"File {self.locked_file_path} not found: {e}")
        except Exception as e:
            logger.error(f"Error occurred while reading {self.locked_file_path}: {e}")

    def __exit__(self, exc_type, exc_value, traceback):
        os.rename(self.locked_file_path, self.file_path)
        if exc_type is not None:
            logging.error(f"An exception occurred within the 'with' block: {exc_type}, {exc_value}")


with LockFile(file_path) as src:
    with open(files_to_blur_full_path, "w") as dest:
        for line in src:
            dest.write(f"{input_structured_folder}/{line}")
```

Reply:
What happened with this?

Reply:
I like this!

Reply:
I got issues with this code. I will open a new branch for that